sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -347,6 +347,14 @@ querySpecification
        windows?)
     ;
 
+hint
+    : '/*+' mapJoinHint '*/'
+    ;
+
+mapJoinHint
+    : MAPJOIN '(' broadcastedTables+=tableIdentifier (',' broadcastedTables+=tableIdentifier)* ')'
```

Comment:
Is the MAPJOIN optimization the only one we are going to support? This is fine if we aren't. We might want to make this more general (and move the logic into the planner) if we are.